core-rust/state-manager/src/store/db.rs

```diff
@@ -87,6 +87,32 @@ use crate::{
 use radix_engine::types::{KeyValueStoreId, SubstateId};
 use radix_engine_stores::hash_tree::tree_store::{NodeKey, Payload, ReadableTreeStore, TreeNode};
 
+// TODO(engine-merge): remove (use the impl in radix engine)
+pub fn encode_substate_id(index_id: &Vec<u8>, db_key: &Vec<u8>) -> Vec<u8> {
```

Comment:
It's not clear to me what this encoding is for - it looks like we will have different encodings for different things now. I'd suggest talking to @jakrawcz-rdx  to get some help with this - as we've discussed this extensively with Josh and have a clearer idea about what's best here 👍